src/cached_object_store.rs

```diff
@@ -53,10 +64,35 @@ impl CachedObjectStore {
             scan_interval,
             db_stats.clone(),
         ));
+
+        // If `max_cache_size_bytes` is None, the initial capacity of ghost queue is 32 entries and capacity will
+        // adjust from 32 to 64 so that we can provide some resistance even if there is no eviction on disk cache
+        //
+        // Otherwise, the initial capacity of ghost queue is `max_cache_size_bytes / 10`, and we will enable
+        // size aware eviction to adaptively provide relevant resistance
+        // TODO(asukamilet): Consider the case where `max_cache_bytes_size` is `Some(0)` which means we disable the disk cache
+        let ghost = max_cache_size_bytes.map_or(
+            Arc::new(Mutex::new(
+                LruBuilder::new(unsafe { NonZeroUsize::new_unchecked(32) }).build(),
+            )),
+            |max_cache_size_bytes| {
+                Arc::new(Mutex::new(
+                    LruBuilder::new(
+                        NonZeroUsize::new(max_cache_size_bytes / 10)
+                            .expect("LRU capacity should not be 0"),
```

Comment:
since this function returns a `Result`, maybe this should not panic and return an appropriate `SlateDbError` variant.

Reply:
My oversight, but I think this is a bit subtle, similar to `max_cache_bytes_size` being Some(0), and I'm not quite sure how to handle ghost queue in this case. according to my discussion with @flaneur2020 , when Some(0) comes up, it means that we don't have disk cache. It's not even an error in this case.

But consider that the user specifies a particularly small `max_cache_bytes_size`, e.g. 1byte. In this case capacity is calculated as 0, but I think it should be considered an error?

Reply:
> I think it should be considered an error?

yes, I agree, it should be considered an error. What I'm suggesting is that the error should be returned to the client, something like `return Err(SlateDbError::ZeroCapacityCache)` instead of panicking.

Reply:
> > I think it should be considered an error?
> 
> yes, I agree, it should be considered an error. What I'm suggesting is that the error should be returned to the client, something like `return Err(SlateDbError::ZeroCapacityCache)` instead of panicking.

+1, such error value looks better for me